core/src/main/java/org/mskcc/cbio/portal/dao/DaoGene.java

```diff
@@ -67,45 +67,58 @@ private static synchronized int getNextFakeEntrezId() throws DaoException {
     public static synchronized int addGeneWithoutEntrezGeneId(CanonicalGene gene) throws DaoException {
         CanonicalGene existingGene = getGene(gene.getHugoGeneSymbolAllCaps());
         gene.setEntrezGeneId(existingGene==null?getNextFakeEntrezId():existingGene.getEntrezGeneId());
-        return addGene(gene);
+        return addOrUpdateGene(gene);
```

Comment:
It is a bit troubling to see updates/development to DaoGene. Since the Dao code is still active, we must make adjustments here I guess, but for each major effort to a Dao class we may want to consider rebuilding the class in a new persistence / service / web manner after the new API under development. We do want to drop the Dao classes (and the core module in general), so these efforts may be wasteful.
Perhaps more important, there is persistence level code in development now (a spring batch importer) based on the existing schemas and relationships between table. Changes such as this will need to be mirrored in that codebase to keep the codebases compatible, so some coordination is needed I think. Also, there is code in the new API for extracting / reporting genes, genetic profiles, etc. That code needs to be updated / coordinated as well. (I expect those change are below in this PR, since there are unit tests written for the new web API endpoints)